### PR TITLE
Fixed project setup scripts

### PR DESCRIPTION
**Pull request by davidz627**
Added some cleanup to project setup scripts, corrected role and added TODO for IAM scope reduction.

/assign @msau42 
Feel free to use updated scripts for yourself too

### CODE REVIEW

File: deploy/setup-project.sh [hunk 1]
```diff
@@ -1,6 +1,17 @@
 #!/bin/bash
 
+set -o nounset
+set -o errexit
+
 IAM_NAME="$GCEPD_SA_NAME@$PROJECT.iam.gserviceaccount.com"
+
+# Cleanup old Service Account and Key
+rm -f $SA_FILE
```
Comment (msau42) [inline deploy/setup-project.sh]: One more recommendation I found was to put quotes around all variables, to handle things like spaces in filenames

Comment (davidz627) [reply deploy/setup-project.sh]: done

File: test/run-tests.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -e
+set -x
+
+readonly PKGDIR=github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver
+
+go test -timeout 30s ${PKGDIR}/test/sanity/ -run ^TestSanity$
+go run $GOPATH/src/${PKGDIR}/test/remote/run_remote/run_remote.go --logtostderr --v 2 --project "${PROJECT}" --zone "${ZONE}" --ssh-env gce --delete-instances=false --cleanup=false --results-dir=my_test
```
Comment (msau42) [inline test/run-tests.sh]: quotes here too :-)

Comment (davidz627) [reply test/run-tests.sh]: done :)

**Pull request merged**